src/source/Crypto/Dtls.c

```diff
@@ -3,6 +3,7 @@
 
 STATUS dtlsSessionOnOutBoundData(PDtlsSession pDtlsSession, UINT64 customData, DtlsSessionOutboundPacketFunc callbackFn)
 {
+    ENTERS();
```

Comment:
Actually, I do not know the difference between ENTER() and ENTERS(), since I can not get the information from code or documents. I just follow the coding inside this sdk, so adding ENTERS(). Would you mind giving me the difference?

Reply:
Oh sure. You can check out PlatformDefs.h in the PIC for more info but ENTER evaluates to DLOGV whereas ENTERS evaluates to DLOGS. 

DLOGV will be enabled when the log level verbosity is set to verbose only.
DLOGS will ONLY be enabled when the verbosity is set to verbose AND it's compiled with LOG_STREAMING compile time def.

LEAVE and LEAVES are the equivalents when leaving the function.

So, we normally use ENTER()/LEAVE() combination for public entries - public APIs that are not super verbose. For example create create or free can have ENTER()/LEAVE() but get metrics for example or put frame might not.

We use ENTERS()/LEAVES() for internal functions which are important to trace but not the super verbose ones. 

We are planning on having a compiler switch that will not only print out ENTERS/LEAVES but will also add trace profiling - for example the non-intrusive trace profiler in PIC

Reply:
Btw, I hope the mechanism of the trace can have the module level. It is easier to do debugging on the specific module.

Reply:
thanks for the detailed information.